src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
```

Comment:
PrimathonAICodeReviewer

Consider adding a constructor to initialize the logger instead of directly initializing it in the class body.
+   constructor(private logger = new Logger()) {}



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->